androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/LauncherActivity.java

```diff
@@ -118,6 +118,14 @@
     @Nullable
     private TwaLauncher mTwaLauncher;
 
+    /**
+     * Signals if `LauncherActivity` should automatically launch the Trusted Web Activity on
+     * `onCreate()`. Set to `false` when a subclass needs to perform an asynchronous task before
+     * launching the Trusted Web Activity. The subclass will then be responsible for calling
+     * `launch()` itself once the asynchronous task is finished.
+     */
+    protected boolean mLaunchImmediately = true;
```

Comment:
I don't like protected variables, usually we can get by with mutable private variables and immutable, non-private variables. (I can probably come up with a more concrete reason I don't like them if you want one.)

I can think of two ways that I think would be better:
1. Create a `boolean shouldLaunchImmediately()` method that can be overridden.
2. Create some protected method like `void onPostCreate()` that gets called at the end of `onCreate` and which by default calls `launchTwa()`. The user can then override this to be empty and call `launchTwa` themselves. (Admittedly the naming here is a bit ugly.)

Reply:
Moved to a `shouldLaunchImmediately()` method.